lib/dynflow/executors/parallel/pool.rb

```diff
@@ -46,12 +46,17 @@ def delete(execution_plan_id)
           end
         end
 
-        def initialize(core, name, pool_size, transaction_adapter)
+        def initialize(world, core, name, pool_size, transaction_adapter)
+          @world = world
           @name = name
           @executor_core = core
           @pool_size     = pool_size
-          @free_workers  = Array.new(pool_size) { |i| Worker.spawn("worker-#{i}", reference, transaction_adapter) }
           @jobs          = JobStorage.new
+          @free_workers  = Array.new(pool_size) do |i|
+            name = "worker-#{i}"
```

Comment:
This name goes to telemetry, it is worth advertising it via UNIX process name so it also appears in `ps ax` output. In case there is some worker struggling this would allow you to identify the process and do something (kill it, finding RSS memory etc). Ideal naming convention would be: `dynlfow [worker-1]` or similar.

Reply:
I'm afraid the workers run as fibers inside the executor process, so all the workers show up just as `dynflow_executor` in `ps ax`

Reply:
Oh ok I thought this is a multi-process environment.